File: core/src/main/scala/diplomacy/rule/ResolvedResult.scala
package diplomacy.rule

import diplomacy.board.{ Power, MilitaryBranch, Board }

final case class ResolvedResult[State_, Power_ <: Power, MilitaryBranch_ <: MilitaryBranch, UnitStatus_, ProvinceStatus_, Order_ <: Order[Power_, MilitaryBranch_], Result_](
  board: Board[State_, Power_, MilitaryBranch_, UnitStatus_, ProvinceStatus_],
  result: Set[OrderResult[Power_, MilitaryBranch_, Order_, Result_]],
  isFinished: Boolean
)

File: src/rule/resolved-result.ts
import { Board } from "./../board/module"
import { OrderResult } from "./order-result"

export class ResolvedResult<Power, MilitaryBranch, State, UnitStatus, ProvinceStatus, Result> {
  /**
   * The set of order results
   */
  results: Set<OrderResult<Power, MilitaryBranch, Result>>
  /**
   * @param board
   * @param results The set of order results
   * @param isFinished The flag whether this game is finished or not.
   */
  constructor (
    public board: Board<Power, MilitaryBranch, State, UnitStatus, ProvinceStatus>,
    results: Set<OrderResult<Power, MilitaryBranch, Result>> | Array<OrderResult<Power, MilitaryBranch, Result>>,
    public isFinished: boolean
  ) {
    this.results = new Set([...results])
  }
}
